### PR TITLE
[query] Don't persist in table._filter_partitions

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
cc @patrick-schultz -- easy fix.

File: hail/python/hail/table.py [hunk 1]
```diff
@@ -3750,7 +3750,7 @@ def summarize(self, handler=None):
 
     @typecheck_method(parts=sequenceof(int), keep=bool)
     def _filter_partitions(self, parts, keep=True) -> 'Table':
-        return Table(ir.TableToTableApply(self._tir, {'name': 'TableFilterPartitions', 'parts': parts, 'keep': keep})).persist()
+        return Table(ir.TableToTableApply(self._tir, {'name': 'TableFilterPartitions', 'parts': parts, 'keep': keep}))
```
Comment (danking) [inline hail/python/hail/table.py:3753]: this seems crazy?

File: hail/python/hail/ir/table_ir.py [hunk 2]
```diff
@@ -934,8 +934,9 @@ def __init__(self, child, config):
 
     def _handle_randomness(self, uid_field_name):
         if uid_field_name is not None:
-            raise FatalError('TableToTableApply does not support randomness in consumers')
-        child = self.child.handle_randomness(None)
+            if self.config['name'] != 'TableFilterPartitions':
+                raise FatalError(f'TableToTableApply({self.config["name"]}) does not support randomness in consumers')
+        child = self.child.handle_randomness(uid_field_name)
```
Comment (danking) [inline hail/python/hail/ir/table_ir.py:939]: why did we switch to using uid_field_name from None?

File: hail/python/hail/ir/table_ir.py [hunk 3]
```diff
@@ -934,8 +934,9 @@ def __init__(self, child, config):
 
     def _handle_randomness(self, uid_field_name):
         if uid_field_name is not None:
-            raise FatalError('TableToTableApply does not support randomness in consumers')
-        child = self.child.handle_randomness(None)
+            if self.config['name'] != 'TableFilterPartitions':
```
Comment (danking) [inline hail/python/hail/ir/table_ir.py:937]: why doers table filter partitions support random consumers?

**Review by danking — CHANGES_REQUESTED** [review]

File: hail/python/hail/table.py (hunk 1 above)
Comment (tpoterba) [reply hail/python/hail/table.py:3753]: the persist was crazy, yes. No persist, no crazy.

File: hail/python/hail/ir/table_ir.py (hunk 2 above)
Comment (tpoterba) [reply hail/python/hail/ir/table_ir.py:939]: TableFilterPartitions is the only TableToTableApply where we do support passing through random UIDs. Others trigger an error here.

File: hail/python/hail/ir/table_ir.py (hunk 3 above)
Comment (tpoterba) [reply hail/python/hail/ir/table_ir.py:937]: why not? super easy to pass through row UIDs.

**Comment by patrick-schultz** [comment]
This looks right to me. I just added persists after all TableToTableApply nodes, so they didn't have to implement uid handling. But Tim is right, FilterPartitions is trivially correct, because it doesn't modify rows.

**Review by danking — APPROVED** [review]

**Pull request merged**